webpack/scenes/ContentViews/Create/CreateContentViewForm.js

```diff
@@ -0,0 +1,154 @@
+import { STATUS } from 'foremanReact/constants';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { Form, FormGroup, TextInput, TextArea, Checkbox, Radio, ActionGroup, Button } from '@patternfly/react-core';
+import { createContentView } from '../ContentViewsActions';
+import { selectCreateContentViews, selectCreateContentViewStatus, selectCreateContentViewError } from './ContentViewCreateSelectors';
+import { labelComponent, labelComposite, labelDependencies, labelAutoPublish } from './CreateContentViewFormComponents';
+
+const CreateContentViewForm = ({ setModalOpen }) => {
+  const dispatch = useDispatch();
+  const [name, setName] = useState('');
+  const [label, setLabel] = useState('');
+  const [description, setDescription] = useState('');
+  const [composite, setComposite] = useState(false);
+  const [component, setComponent] = useState(true);
+  const [autoPublish, setAutoPublish] = useState(false);
+  const [importOnly, setImportOnly] = useState(false);
+  const [dependencies, setDependencies] = useState(false);
+  const [redirect, setRedirect] = useState(false);
+  const [redirectUrl, setRedirectURL] = useState('/labs/content_views');
```

Comment:
Take a look at the `ForemanForm` component: http://foreman-storybook.surge.sh/?path=/story/components-foreman-form--basic-foreman-form

It uses a library called [Formik](https://formik.org/) and it may help reduce some of this boilerplate for keeping track of the form fields' state.

Reply:
@jeremylenz so `ForemanForm` would wrap the pf4 input components? 

If so, I think it would be ok, the only built-in pf3 component I see is the alert [here](https://github.com/theforeman/foreman/blob/c18d67464d81943b77feeb0e24ec649574860ab7/webpack/assets/javascripts/react_app/components/common/forms/Form.js#L3), ideally we upgrade it but its fine for the time being IMO

Reply:
@johnpmitsch Yeah I don't see any issue with having PF4 components as children of the form.  `ForemanModal` is a little bit more tricky because it does have the PF3 modal components baked in..

Reply:
Another consideration is I'm not sure if ForemanForm has been updated to the newer version of Formik that works well with hooks.  probably should check that before going too far.

Reply:
I am not entirely sure that ForemanForm is handling pf4 components well as children..It only seems to work if I use TextField from foreman forms (https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/forms/TextField/TextField.js).
Otherwise it doesn't tie the input to formik state which seems to be expected since we don't have formik wrappers for the pf4 form components AFAIU. Does that make sense?

Reply:
Ok, looks like the existing Foreman components aren't keeping up.  Per our discussion just now I guess we should go with the current approach for now and discuss how to upgrade ForemanForm later.